crates/db_views_actor/src/community_view.rs

```diff
@@ -203,8 +204,8 @@ impl<'a> CommunityQuery<'a> {
       query = query.filter(community_block::person_id.is_null());
       query = query.filter(community::nsfw.eq(false).or(local_user::show_nsfw.eq(true)));
     } else {
-      // No person in request, only show nsfw communities if show_nsfw passed into request
-      if !self.local_user.map(|l| l.show_nsfw).unwrap_or(false) {
+      // No person in request, only show nsfw communities if show_nsfw is passed into request
+      if !self.show_nsfw.unwrap_or(false) {
```

Comment:
What you replaced should be illuminating. 

You are overriding the users show_nsfw preference. Why?

Reply:
![image](https://github.com/LemmyNet/lemmy/assets/293107/7cbea872-4388-46f2-b05f-c48bde58265e)
Does not the first part of this condition check where there IS a local user, the `else` looks like it should be when there is no local user.

I'm not full bottle on the syntax in rust. I did test this using docker-compose and it still appears to respect my show_nsfw preference when logged in. _(and the `snow_nsfw` query param appeared to work)_

Reply:
Yep this looks correct, it makes no sense to check for local user in else branch.

Reply:
Ahhh okay I see that now, sry.